org.eclipse.jdt.core/search/org/eclipse/jdt/core/search/SearchPattern.java

```diff
@@ -2453,10 +2453,19 @@ private static char[][] enclosingTypeNames(IType type) {
 		case IJavaElement.FIELD:
 		case IJavaElement.INITIALIZER:
 		case IJavaElement.METHOD:
+			char[] typeName = IIndexConstants.ONE_STAR;
+			try {
+				String superclassName = type.getSuperclassName();
```

Comment:
I would really like if you can add some code comments describing what we are doing here and why for future us or another developer :)